### ISSUE TITLE
Permanent diff for `google_bigquery_dataset`s `max_time_travel_hours`

### ISSUE THREAD

**Issue by multani** [hashicorp/terraform-provider-google#13689]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
$ terraform -v      
Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.52.0
+ provider registry.terraform.io/hashicorp/google-beta v4.52.0
```

### Affected Resource(s)

* `google_bigquery_dataset`

### Terraform Configuration Files

```tf
resource "google_bigquery_dataset" "metering" {
  project = data.google_project.this.project_id

  dataset_id = "gke_usage_metering_dataset"

  friendly_name = "GKE usage"
  description   = "Measurements of our GKE clusters' usage"

  location = "EU"
}
```

### Debug Output

https://gist.github.com/multani/e69e02a87340868b9c45347bd0c81d1c

### Expected Behavior

Once created, this resource should not be changed by Terraform.

### Actual Behavior

Terraform tries to remove the `max_time_travel_hours` attribute:

```hcl
Terraform will perform the following actions:

  # google_bigquery_dataset.metering will be updated in-place
  ~ resource "google_bigquery_dataset" "metering" {
        id                              = "projects/camunda-saas-dev/datasets/gke_usage_metering_dataset"
      - max_time_travel_hours           = "168" -> null
        # (13 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Reapplying again produces the same diff over and over.

Setting a specific value for `max_time_travel_hours` masks the problem.

### Steps to Reproduce

1. `terraform apply`

### References

* #12830

**Comment by edwardmedia** [comment]
@multani I don't see `max_time_travel_hours` in your config . To confirm, you previously set it with 168 and then try to remove it?

**Comment by multani** [comment]
@edwardmedia No, I never set it before, it started to appear in the `terraform plan` output I think after v4.52.0.

168 is I think the default value when you don't specify anything.

**Comment by edwardmedia** [comment]
@multani I do see it now with the latest version. Thanks

**Comment by multani** [comment]
@edwardmedia any chance to get the fix for this issue released soon?
